### PR TITLE
Updating Storage Files swagger to 2019-02-02

### PR DESCRIPTION
**Pull request by tg-msft**
This includes a few transform changes to set default values for new required
parameters and marked certain 8601 dates as strings when they use more precision
that Python's dates.

I'm seeing a few test failures that I'm hoping someone on the Python side can
investigate _(mostly "The specified share is being deleted. Try operation later."
that I was seeing before my changes)_.  There are a few other failures related to
new service features that we can fix if easy or disable and file a work item to
get these changes in.  The recorded tests will also have to be updated, but we
don't need to block on that right now.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by rakshith91** [comment]
CI is green :o

**Comment by annatisch** [comment]
LGTM - thanks @tg-msft
@zezha-msft  - there a number of "TODOs" in the code to confirm whether the defaults used for values are correct. Are you able to weigh in here?
(Specifically they are in the swagger readme, and the file_client/file_client_async)

File: sdk/storage/azure-storage-file/swagger/README.md [hunk 1]
```diff
@@ -48,4 +48,51 @@ directive:
...
+
+### FileRangeWriteFromUrl Constant
+``` yaml
+directive:
+- from: swagger-document
+  where: $.parameters.FileRangeWriteFromUrl
+  transform: >
+    delete $.default;
+    delete $["x-ms-enum"];
+    $["x-ms-parameter-location"] = "method";
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-file/swagger/README.md]: please add a comment explaining this change

**Comment by zezha-msft** [comment]
The default values look good to me. @xiafu-msft please double check them since you did the feature for track 1.

**Comment by zezha-msft** [comment]
>                 "parameters": [{

Where is the body (permission string) specified?

---
Refers to: sdk/storage/azure-storage-file/swagger/file-2019-02-02.json:519 in 3129e36. [](commit_id = 3129e363776065513dc1f3b1f141d9006f25d7df, deletion_comment = False)

File: sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_share_operations.py [hunk 2]
```diff
@@ -293,6 +293,123 @@ def create_snapshot(self, timeout=None, metadata=None, cls=None, **kwargs):
             return cls(response, None, response_headers)
     create_snapshot.metadata = {'url': '/{shareName}'}
 
+    def create_permission(self, timeout=None, cls=None, **kwargs):
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_share_operations.py]: >def create_permission(self, timeout=None, cls=None, **kwargs): [](start = 4, length = 62)

It seems the swagger is not correct? the permission should be in request body, currently it looks like the quest body is empty?

Comment (tg-msft) [reply sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_share_operations.py]: Correct - your current swagger files don't have the JSON permission body.

File: sdk/storage/azure-storage-file/swagger/README.md (hunk 1 above)
Comment (tg-msft) [reply sdk/storage/azure-storage-file/swagger/README.md]: This value is supposed to be the constant value `update` and these changes turn it from a parameter into a constant.  We'll update with a comment.

File: sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_share_operations.py (hunk 2 above)
Comment (zezha-msft) [reply sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_share_operations.py]: @tg-msft thanks for the clarification! We can still pass in the serialized body(done through hand-written code?), right?

Comment (tg-msft) [reply sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_share_operations.py]: I'm not sure -- @annatisch could answer that better.

My team is actively investigating if we can have it generate the JSON serialization for you _(with a future swagger update, of course)_.  I'd suggest you hold off on the CreatePermission/GetPermission APIs like .NET is doing until we can loop back around with you on that.

File: sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_file_operations.py [hunk 3]
```diff
@@ -394,6 +462,17 @@ def set_http_headers(self, timeout=None, file_content_length=None, file_http_hea
          the specified byte value is less than the current size of the file,
          then all ranges above the specified byte value are cleared.
         :type file_content_length: long
+        :param file_permission: If specified the permission (security
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_file_operations.py]: >file_permission [](start = 15, length = 15)

It seems the default value for those new added parameters should be 'Preserve'?

File: sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_file_operations.py [hunk 4]
```diff
@@ -382,9 +440,19 @@ def delete(self, timeout=None, cls=None, **kwargs):
             return cls(response, None, response_headers)
     delete.metadata = {'url': '/{shareName}/{directory}/{fileName}'}
 
-    def set_http_headers(self, timeout=None, file_content_length=None, file_http_headers=None, cls=None, **kwargs):
+    def set_http_headers(self, file_attributes="none", file_creation_time="now", file_last_write_time="now", timeout=None, file_content_length=None, file_permission="inherit", file_permission_key=None, file_http_headers=None, cls=None, **kwargs):
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_file_operations.py]: It seems the default value for those new added parameters should be 'Preserve'?

Comment (tg-msft) [reply sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_file_operations.py]: Your swagger shares the same parameter definition for Create and SetProperties.  The default value for Create is `Now` and the default value for SetProperties is `Preserve`.  We can patch up the swagger at some point, but for now we're passing `Preserve` in the methods that call SetProperties.

File: sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_directory_operations.py [hunk 5]
```diff
@@ -213,6 +255,98 @@ def delete(self, timeout=None, cls=None, **kwargs):
             return cls(response, None, response_headers)
     delete.metadata = {'url': '/{shareName}/{directory}'}
 
+    def set_properties(self, file_attributes="none", file_creation_time="now", file_last_write_time="now", timeout=None, file_permission="inherit", file_permission_key=None, cls=None, **kwargs):
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_directory_operations.py]: Also for this one, would you like to verify that we don't want to use 'Preserve' as default value? :blush:

File: sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_share_operations.py (hunk 2 above)
Comment (zezha-msft) [reply sdk/storage/azure-storage-file/azure/storage/file/_generated/operations/_share_operations.py]: Sounds good. Let's merge as is, and fix up the REST parity work later

**Pull request merged**